certs/cert_pool_nix.go

```diff
@@ -35,52 +33,14 @@ var certDirectories = []string{
 	"/var/run/secrets/kubernetes.io/serviceaccount",
 }
 
-// readUniqueDirectoryEntries is like ioutil.ReadDir but omits
-// symlinks that point within the directory.
-func readUniqueDirectoryEntries(dir string) ([]os.FileInfo, error) {
-	fis, err := ioutil.ReadDir(dir)
-	if err != nil {
-		return nil, err
-	}
-	uniq := fis[:0]
-	for _, fi := range fis {
-		if !isSameDirSymlink(fi, dir) {
-			uniq = append(uniq, fi)
-		}
-	}
-	return uniq, nil
-}
-
-// isSameDirSymlink reports whether fi in dir is a symlink with a
-// target not containing a slash.
-func isSameDirSymlink(fi os.FileInfo, dir string) bool {
-	if fi.Mode()&os.ModeSymlink == 0 {
-		return false
-	}
-	target, err := os.Readlink(filepath.Join(dir, fi.Name()))
-	return err == nil && !strings.Contains(target, "/")
-}
-
 func loadSystemRoots() (*x509.CertPool, error) {
+	// Add additional ENV to load k8s CA certs.
+	os.Setenv("SSL_CERT_DIR", strings.Join(certDirectories, ":"))
```

Comment:
Maybe you can append instead of overwriting this env variable. It is good to have this env variable configurable by the user.

Reply:
Yeah can do that @vadmeste